frame/staking-rewards/src/lib.rs

```diff
@@ -85,11 +132,150 @@ pub mod pallet {
 		#[pallet::constant]
 		type PalletId: Get<PalletId>;
 
+		/// Maximum number of staking duration presets allowed.
+		#[pallet::constant]
+		type MaxStakingDurationPresets: Get<u32>;
+
+		/// Maximum number of reward configurations per pool.
+		#[pallet::constant]
+		type MaxRewardConfigsPerPool: Get<u32> + TypeInfo;
+
+		/// Required origin for reward pool creation.
+		type RewardPoolCreationOrigin: EnsureOrigin<Self::Origin>;
+
 		type WeightInfo: WeightInfo;
 	}
 
+	/// Abstraction over RewardPoolConfiguration type
+	type RewardPoolConfigurationOf<T> = RewardPoolConfiguration<
+		<T as frame_system::Config>::AccountId,
+		<T as Config>::MayBeAssetId,
+		<T as Config>::Balance,
+		<T as frame_system::Config>::BlockNumber,
+		StakingDurationToRewardsMultiplierConfig<<T as Config>::MaxStakingDurationPresets>,
+	>;
+
+	/// Abstraction over RewardPool type
+	type RewardPoolOf<T> = RewardPool<
+		<T as frame_system::Config>::AccountId,
+		<T as Config>::MayBeAssetId,
+		<T as Config>::Balance,
+		<T as frame_system::Config>::BlockNumber,
+		StakingDurationToRewardsMultiplierConfig<<T as Config>::MaxStakingDurationPresets>,
+		<T as Config>::MaxRewardConfigsPerPool,
+	>;
+
 	#[pallet::pallet]
-	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::generate_store(pub (super) trait Store)]
 	#[pallet::without_storage_info]
 	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	#[pallet::getter(fn pool_count)]
+	#[allow(clippy::disallowed_types)]
+	pub type RewardPoolCount<T: Config> =
+		StorageValue<_, T::PoolId, ValueQuery, Counter<ZeroInit, SafeIncrement, SafeDecrement>>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn pools)]
+	pub type RewardPools<T: Config> = StorageMap<_, Blake2_128Concat, T::PoolId, RewardPoolOf<T>>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a new reward pool based on the config.
+		///
+		/// Emits `RewardPoolCreated` event when successful.
+		#[pallet::weight(T::WeightInfo::create_reward_pool())]
+		#[transactional]
+		pub fn create_reward_pool(
+			origin: OriginFor<T>,
+			pool_config: RewardPoolConfigurationOf<T>,
+		) -> DispatchResult {
+			T::RewardPoolCreationOrigin::ensure_origin(origin)?;
+			let (owner, pool_id, end_block) = match pool_config {
+				RewardRateBasedIncentive {
+					owner,
+					asset_id,
+					initial_reward_config,
+					end_block,
+					lock,
+				} => {
+					ensure!(
+						end_block > frame_system::Pallet::<T>::current_block_number(),
+						Error::<T>::InvalidEndBlock
```

Comment:
please be more specific in error.
please use Validated from configuration.

Reply:
please use UnixTime trate as dep like in other pallets

Reply:
Not really sure about the `UnixTime` trait suggestion, could you link to an example? Here we don't need time

Reply:
it was team decision to use Validated. if you want to override please ensure you settle that decision.  you validate big configuration. 

example, please validate asset_id exists in extinsic call.

Reply:
https://app.clickup.com/t/2n9rhpd

Reply:
please get approve not using validate as it was decision of team and leads before.

Reply:
please estimate that task if need too